File: src/LiWiMus.API/Areas/User/Controllers/UserController.fs
namespace LiWiMus.API.Areas.User.Controllers

open System.Threading.Tasks
open AutoMapper
open LiWiMus.API.Areas.User.Dto
open LiWiMus.Core.Users
open LiWiMus.Core.Users.Specifications
open LiWiMus.SharedKernel.Interfaces
open Microsoft.AspNetCore.Identity
open Microsoft.AspNetCore.Mvc

[<ApiController>]
[<Route("api/[area]/[controller]/[action]")>]
type UserController(mapper: IMapper,
                    userManager: UserManager<User>,
                    userRepository: IRepository<User>) =
    inherit ControllerBase()
    
//    [<HttpPost>]
//    member this.Create(userDto: UserDtoCreate) : Task<IActionResult> =
//        async {
//            let! user = userDto.UserName |> userManager.FindByNameAsync |> Async.AwaitTask
//            if user |> isNull = false then
//                return ConflictResult()
//            else
//                user = mapper.Map<User>(userDto)
//                let! result = userManager.CreateAsync(user, userDto.Password) |> Async.AwaitTask
//                if result.Succeeded then
//                    let createdUser = userRepository.GetBySpecAsync(UserWithIncludes(user, nameof(user.Artists)))
//                    let createdUserDto = mapper.Map<UserDtoReadOne>(user)
//                    return CreatedResult("", user)
//                else
//                    return 
//        } |> Async.StartAsTask

File: src/LiWiMus.Web.MVC/Areas/Search/Controllers/UserController.cs
using AutoMapper;
using LiWiMus.Core.Shared;
using LiWiMus.Core.Users.Specifications;
using LiWiMus.SharedKernel.Interfaces;
using LiWiMus.Web.MVC.Areas.Search.ViewModels;
using LiWiMus.Web.MVC.ViewModels.ForListViewModels;
using Microsoft.AspNetCore.Mvc;

namespace LiWiMus.Web.MVC.Areas.Search.Controllers;

[Area(AreasConstants.Search)]
public class UserController : Controller
{
    private readonly IRepository<Core.Users.User> _userRepository;
    private readonly IMapper _mapper;

    public UserController(IRepository<Core.Users.User> userRepository, IMapper mapper)
    {
        _userRepository = userRepository;
        _mapper = mapper;
    }

    private async Task<IEnumerable<UserForListViewModel>> GetUsersAsync(SearchViewModel searchVm)
    {
        var pagination = _mapper.Map<Pagination>(searchVm);
        var users = await _userRepository.PaginateWithTitleAsync(pagination);
        return _mapper.Map<List<Core.Users.User>, List<UserForListViewModel>>(users);
    }
    
    [HttpGet]
    public async Task<IActionResult> Index()
    {
        var users = await GetUsersAsync(SearchViewModel.Default);
        return View(users);
    }

    [HttpGet]
    public async Task<IActionResult> ShowMore(SearchViewModel searchVm)
    {
        var users = await GetUsersAsync(searchVm);
        return PartialView("UsersPartial", users);
    }
}